tests/lib/prepare.sh

```diff
@@ -243,10 +246,9 @@ prepare_classic() {
         for unit in $units; do
             systemctl stop "$unit"
         done
-        snapd_env="/etc/environment /etc/systemd/system/snapd.service.d /etc/systemd/system/snapd.socket.d"
-        snap_confine_profiles="$(ls /etc/apparmor.d/snap.core.* || true)"
-        # shellcheck disable=SC2086
-        tar cf "$SPREAD_PATH"/snapd-state.tar.gz /var/lib/snapd "$SNAP_MOUNT_DIR" /etc/systemd/system/"$escaped_snap_mount_dir"-*core*.mount /etc/systemd/system/multi-user.target.wants/"$escaped_snap_mount_dir"-*core*.mount $snap_confine_profiles $snapd_env
+
+        save_snapd_state "$escaped_snap_mount_dir"
```

Comment:
Just a nitpick, I find the use of the escaped path unusual.

Reply:
this is not done anymore, now we are calculating escaped_snap_mount_dir as part of the state script so there is not any parameters needed.